src/Microsoft.Data/BaseColumn.cs

```diff
@@ -65,9 +65,7 @@ protected set
         /// <param name="mapIndices"></param>
         /// <param name="invertMapIndices"></param>
         /// <returns></returns>
-        public virtual BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapIndices = false) => throw new NotImplementedException();
-
-        public virtual BaseColumn CloneAndAppendNulls(BaseColumn mapIndices = null, bool invertMapIndices = false) => throw new NotImplementedException();
+        public virtual BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapIndices = false, long numberOfNullsToAppend = 0) => throw new NotImplementedException();
```

Comment:
Can `numberOfNullsToAppend` be inferred by the column being cloned? Or does it always have to be specified?

Reply:
Nope, it cannot be inferred. That was my first instinct as well. Sometimes(like in joins and merges), it depends on the number of rows in another table.